src/pybind/mgr/dashboard_v2/tests/test_notification.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import random
+import time
+import unittest
+
+
+from ..tools import NotificationQueue
+
+
+class Listener(object):
+    def __init__(self):
+        NotificationQueue.register(self.log_type1, 'type1')
+        NotificationQueue.register(self.log_type2, 'type2')
+        NotificationQueue.register(self.log_type1_3, ['type1', 'type3'])
+        NotificationQueue.register(self.log_all)
+        self.type1 = []
+        self.type2 = []
+        self.type1_3 = []
+        self.all = []
+
+        # these should be ignored by the queue
+        NotificationQueue.register(self.log_type1, 'type1')
+        NotificationQueue.register(self.log_type1_3, ['type1', 'type3'])
+        NotificationQueue.register(self.log_all)
+
+    def log_type1(self, val):
+        self.type1.append(val)
+
+    def log_type2(self, val):
+        self.type2.append(val)
+
+    def log_type1_3(self, val):
+        self.type1_3.append(val)
+
+    def log_all(self, val):
+        self.all.append(val)
+
+    def clear(self):
+        self.type1 = []
+        self.type2 = []
+        self.type1_3 = []
+        self.all = []
+
+
+class NotificationQueueTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.listener = Listener()
+
+    def setUp(self):
+        self.listener.clear()
+
+    def test_invalid_register(self):
+        # pylint: disable=broad-except
+        try:
+            NotificationQueue.register(None, 1)
+        except Exception as ex:
+            self.assertEqual(str(ex), "types param is neither a string nor a "
+                                      "list")
+
+    def test_notifications(self):
+        NotificationQueue.start_queue()
+        NotificationQueue.new_notification('type1', 1)
+        NotificationQueue.new_notification('type2', 2)
+        NotificationQueue.new_notification('type3', 3)
+        NotificationQueue.stop()
+        self.assertEqual(self.listener.type1, [1])
+        self.assertEqual(self.listener.type2, [2])
+        self.assertEqual(self.listener.type1_3, [1, 3])
+        self.assertEqual(self.listener.all, [1, 2, 3])
+
+    def test_notifications2(self):
+        NotificationQueue.start_queue()
+        for i in range(0, 600):
```

Comment:
Would it reduce your confidence, if we change this to `for i in range(0, 60):`?

Reply:
I'm trying to guarantee that there is enough traffic on the queue to cover some special concurrency case, for instance multiple adds concurrent with notify the listeners.
The way it is now it runs in less than a second in my laptop, do you think it will take much longer in other systems?

Reply:
nope. I agree